3sum.java

```diff
@@ -1,3 +1,12 @@
+// Time Complexity : nlogn + n*n = O(n^2)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : I faced many difficulties with this, my main problem was 
+// figuring out the two pointer approach after 1 element was fixed.
+
```

Comment:
Its okay !!!
Keep practicing 2-pointer technique. This approach comes in handy in sorted Matrix questions as well